packages/console/src/pages/ApplicationDetails/SamlApplicationDetailsContent/Settings.tsx

```diff
@@ -84,7 +90,49 @@ function Settings({ data, mutateApplication, isDeleted }: Props) {
     })
   );
 
-  const secretTableColumns = useSecretTableColumns({ appId: data.id });
+  const onDelete = useCallback(
+    async (id: string) => {
+      await api.delete(`api/saml-applications/${data.id}/secrets/${id}`);
+      toast.success(t('application_details.secrets.deleted'));
+      void secrets.mutate((secrets.data ?? []).filter(({ id: secretId }) => secretId !== id));
+    },
+    [api, data.id, secrets, t]
+  );
+
+  const onActivate = useCallback(
+    async (id: string) => {
+      await api.patch(`api/saml-applications/${data.id}/secrets/${id}`, { json: { active: true } });
+      toast.success(t('application_details.secrets.activated'));
+      void secrets.mutate(
+        (secrets.data ?? []).map((secret) =>
+          secret.id === id ? { ...secret, active: true } : secret
+        )
+      );
+    },
+    [api, data.id, secrets, t]
+  );
+
+  const onDeactivate = useCallback(
+    async (id: string) => {
+      await api.patch(`api/saml-applications/${data.id}/secrets/${id}`, {
+        json: { active: false },
+      });
+      toast.success(t('application_details.secrets.deactivated'));
+      void secrets.mutate(
+        (secrets.data ?? []).map((secret) =>
+          secret.id === id ? { ...secret, active: false } : secret
+        )
+      );
+    },
+    [api, data.id, secrets, t]
+  );
```

Comment:
nit: These two can be combined as a `toggleActivision` function that accepts a boolean flag

Reply:
Use a separate API to handle this since the status change on the current secret could also have effect on others.